client/parser/format.go

```diff
@@ -0,0 +1,748 @@
+package parser
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type varDefState int
+
+const (
+	None     = iota // not currently in a var def
+	Defining        // currently formatting a var def (cannot go to new line)
+	LastNode        // last node of a var def, treated specially
+)
+
+type formatter struct {
+	softWrapLen int         // configured line length to soft wrap formatting
+	indentText  string      // configured indent string (i.e. spaces vs tabs)
+	varDef      varDefState // state to handle formatting variable definitions
+	indentLevel int         // state for indentation level
+	texts       []string    // buffer of "tokens" for the ongoing formatted line
+	out         io.Writer
+}
+
+type formatterOption func(*formatter)
+
+func newFormatter(out io.Writer, opts ...formatterOption) *formatter {
+	formatter := &formatter{
+		// TODO: make these configurable from cli
+		softWrapLen: 80,
+		indentText:  "    ",
```

Comment:
For now, let's make this 2 spaces by default. Personal preference :)   To your point above, we could eventually make this configurable.

Reply:
Changed to 2 spaces default. Also updated the cmd to make those two configurable.